### ISSUE TITLE
Large files on Mac are not getting chunked

### ISSUE THREAD

**Issue by joecoder1021** [mar10/wsgidav#91]
When copying a large file into a webdav folder, the entire file is being read into memory before being written out.

This happens when HTTP_TRANSFER_ENCODING is set to "chunked".

This is being checked in request_server.py:
```py
if environ.get("HTTP_TRANSFER_ENCODING", "").lower() == "chunked":
    data_stream = self._streamDataChunked(environ)
else:
    data_stream = self._streamData(environ, contentlength, self.block_size)
```
In _streamDataChunked(), the logic does not set a maximum chunk size.
The maximum chunk size ends up being the entire size of the file.
It seems that there should be a way to specify the maximum chunk size.

This is around line 639 in request_server.py:
```py
while l > 0:
    buf = environ["wsgi.input"].read(l)
```
Could we add a maximum chunk size parameter just like _StreamData()?

**Comment by mar10** [comment]
Makes sense, thanks.

**Comment by joecoder1021** [comment]
Should I try do to a pull request for this?

**Comment by mar10** [comment]
That would certainly help, especially since you already have a test case. Thanks for contributing!

**Comment by mar10** [comment]
Closed by #93, thanks
